src/main/java/com/spikes2212/state/LayoutManager.java

```diff
@@ -0,0 +1,36 @@
+package com.spikes2212.state;
+
+import edu.wpi.first.wpilibj2.command.Command;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class LayoutManager<T extends Enum<T>> {
```

Comment:
In my opinion, it is better for `LayoutManager` to receive two generic types - `T` for the state type and `K` for the `ButtonLayout` type.